source/windows/secure_channel_tls_handler.c

```diff
@@ -1039,9 +1039,22 @@ static int s_do_application_data_decrypt(struct aws_channel_handler *handler) {
                 read_len);
             sc_handler->buffered_read_in_data_buf.len = read_len;
             aws_raise_error(AWS_IO_READ_WOULD_BLOCK);
+        }
+        /* SEC_I_CONTEXT_EXPIRED means that the message sender has shut down the connection.  One such case
+           where this can happen is an unaccepted certificate. */
+        else if (status == SEC_I_CONTEXT_EXPIRED) {
+            AWS_LOGF_TRACE(
+                AWS_LS_IO_TLS,
+                "id=%p: Message sender has shut down the connection. SECURITY_STATUS is %d.",

```

Comment:
If you can find a way to get the alert code and log it that would be super helpful, if you can’t at least log an alert was recieved.